src/xunit.analyzers/Utilities/TypeHierarchyComparer.cs

```diff
@@ -12,9 +12,9 @@ internal class TypeHierarchyComparer : IComparer<ITypeSymbol>
 
         public int Compare(ITypeSymbol x, ITypeSymbol y)
         {
-            if (x == null || x.TypeKind != TypeKind.Class)
+            if (x?.TypeKind != TypeKind.Class)
                 throw new ArgumentException("The argument must be a class", nameof(x));
-            if (y == null || x.TypeKind != TypeKind.Class)
```

Comment:
Spotted a bug here while I was refactoring: it should be `y.TypeKind != TypeKind.Class`